omod/src/test/java/org/openmrs/module/idgen/web/controller/LogEntryControllerTest.java

```diff
@@ -0,0 +1,163 @@
+/* * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.idgen.web.controller;
+
+import org.junit.Test;
+import org.junit.Before;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.idgen.service.IdentifierSourceService;
+import org.openmrs.module.webservices.rest.SimpleObject;
+import org.openmrs.module.webservices.rest.test.Util;
+import org.openmrs.module.webservices.rest.web.v1_0.controller.MainResourceControllerTest;
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.web.bind.annotation.RequestMethod;
+
+import org.junit.Assert;
+
+public class LogEntryControllerTest extends MainResourceControllerTest {
+
+    private IdentifierSourceService service;
+
+    @Before
+    public void setUp() throws Exception {
+        executeDataSet("org/openmrs/module/idgen/include/TestData.xml");
+        this.service = Context.getService(IdentifierSourceService.class);
+    }
+
+    @Override
+    public long getAllCount() {
+        return service.getLogEntries(null, null, null, null, null, null).size();
+    }
+
+    @Override
+    public String getURI() {
+        return "idgen/logentry";
+    }
+
+    @Override
+    public String getUuid() {
+        return "100892";
+    }
+
+    @Override
+    public void shouldGetAll() throws Exception {
+        super.shouldGetAll();
+    }
+
+    @Test
+    public void shouldListAllLogEntries() throws Exception {
+        MockHttpServletRequest req = request(RequestMethod.GET, getURI());
+        SimpleObject result = deserialize(handle(req));
+        Assert.assertNotNull(result);
+        Assert.assertEquals(getAllCount(), Util.getResultsSize(result));
+    }
+
+    @Test
+    public void shouldSearchAndReturnAListOfLogEntriesWithGivenSourceUuid() throws Exception {
+        MockHttpServletRequest req = request(RequestMethod.GET, getURI());
+        req.addParameter("source", "0d47284f-9e9b-4a81-a88b-8bb42bc0a901");
```

Comment:
declare constants at the top of the file for the source, generatedBy uuid's

Reply:
Fixed